crates/holochain/src/conductor/interface/websocket.rs

```diff
@@ -65,22 +65,25 @@ pub fn spawn_admin_interface_tasks<A: InterfaceApi>(
 
     tm.add_conductor_task_ignored(&format!("admin interface, port {}", port), |_stop| {
         async move {
-            let num_connections = Arc::new(AtomicIsize::new(0));
+            let mut active_connections = Vec::new();
             futures::pin_mut!(listener);
             // establish a new connection to a client
             while let Some(connection) = listener.next().await {
+                active_connections.retain_mut(|handle: &mut JoinHandle<()>| !handle.is_finished());
+
                 match connection {
                     Ok((_, rx_from_iface)) => {
-                        if num_connections.fetch_add(1, Ordering::Relaxed) > MAX_CONNECTIONS {
```

Comment:
Increment and continue means that we could have a 'phantom' count here that never decreased. This is how you end up with an unresponsive admin websocket (at least one possible way). I can't find any issues with handles not releasing when the other side disconnects or dies but this problem accounting very quickly causes problems because every connection attempt when `num_connections > MAX_CONNECTIONS` increments `num_connections` and after a `MAX_CONNECTIONS` attempts, it won't matter how many clients disconnect the system won't recover, you have to restart the conductor.